pkg/daemon/update.go

```diff
@@ -426,17 +428,28 @@ func (dn *Daemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newConfig
 
 }
 
-func calculatePostConfigChangeActionFromFileDiffs(oldIgnConfig, newIgnConfig ign3types.Config) (actions []string) {
+func getPostConfigChangeAction(path string) string {
 	filesPostConfigChangeActionNone := []string{
 		"/etc/kubernetes/kubelet-ca.crt",
 		"/var/lib/kubelet/config.json",
 	}
+
 	filesPostConfigChangeActionReloadCrio := []string{
+		containerRegistryConfPath,
 		GPGNoRebootPath,
 		"/etc/containers/policy.json",
-		"/etc/containers/registries.conf",
 	}
 
+	if ctrlcommon.InSlice(path, filesPostConfigChangeActionNone) {
```

Comment:
I think this might be easier to read without the `else if` conditions:

```golang
if ctrlcommon.InSlice(path, filesPostConfigChangeActionNone) {
  return postConfigChangeActionNone
}

if ctrlcommon.InSlice(path, filesPostConfigChangeActionReloadCrio) {
  return postConfigChangeActionReloadCrio
}

return postConfigChangeActionReboot
```